compiler/resolution/wrappers.cpp

```diff
@@ -1592,6 +1592,14 @@ static void addArgCoercion(FnSymbol*  fn,
   }
 
   resolveCall(castMove);
+
+  IntentTag concreteIntent = concreteIntentForArg(formal);
+  bool formalIsRef = formal->isRef() || (concreteIntent & INTENT_REF);
+
+  if (!castTemp->isRef() && formalIsRef) {
+    USR_FATAL_CONT(call, "Cannot coerce argument for const ref intent");
+  }
```

Comment:
I think it'd be better to fold this into a checking function that includes/replaces calls to argumentCanModifyActual/issueErrorValueTemporaryToRef